### PR TITLE
Fix BlockProperty

### PR DESCRIPTION
**Pull request by ExE-Boss**
This PR fixes Soni's weirdness from #226 (which would cause a `NullPointerException` and an `IllegalArgumenException` once my assets rewrite would be fully implemented) and implements `BlockProperty.Replaceable`.
It also fixes block place sounds.

### CODE REVIEW

File: minecraft/1.8/src/main/java/nova/core/wrapper/mc/forge/v18/launcher/NovaMinecraft.java [hunk 1]
```diff
@@ -190,6 +190,7 @@ public void preInit(FMLPreInitializationEvent evt) {
 				fmlProgressBar.step(wrapper.getClass());
 				wrapper.preInit(evt);
 			});
+			fmlProgressBar.finish();
```
Comment (RX14) [inline minecraft/1.8/src/main/java/nova/core/wrapper/mc/forge/v18/launcher/NovaMinecraft.java]: Seperate this into another PR.

File: minecraft/1.8/src/main/java/nova/core/wrapper/mc/forge/v18/wrapper/block/forward/FWBlock.java [hunk 2]
```diff
@@ -312,7 +328,7 @@ public int getLightValue(IBlockAccess access, BlockPos pos) {
 		Optional<LightEmitter> opEmitter = blockInstance.components.getOp(LightEmitter.class);
 
 		if (opEmitter.isPresent()) {
-			return Math.round(opEmitter.get().emittedLevel.get() * 15.0F);
+			return (int) Math.min(0, Math.max(Math.round(opEmitter.get().emittedLevel.getAsDouble() * 15), 15));
```
Comment (RX14) [inline minecraft/1.8/src/main/java/nova/core/wrapper/mc/forge/v18/wrapper/block/forward/FWBlock.java]: Use `MathUtil.clamp`

File: src/main/java/nova/core/block/component/BlockProperty.java [hunk 3]
```diff
@@ -85,8 +85,8 @@ public double getResistance() {
 	 */
 	@UnsidedComponent
 	public static class BlockSound extends Component implements BlockProperty {
-		public BiMap<BlockSoundTrigger, Sound> blockSoundSoundMap = HashBiMap.create();
-		public Optional<BiMap<String, Sound>> customDefinedSounds;
+		private final Map<BlockSoundTrigger, Sound> blockSounds = new HashMap<>();
+		private Optional<Map<String, Sound>> customSounds = Optional.empty();
```
Comment (RX14) [inline src/main/java/nova/core/block/component/BlockProperty.java]: Do we need to distinguish between `Optional.empty` and an empty map here? If so, just use the latter and simplify the logic massively.

Comment (ExE-Boss) [inline src/main/java/nova/core/block/component/BlockProperty.java]: That's really just Soni's weirdness, again.

**Pull request merged**